### ISSUE TITLE
UrlGenerator::setRouteCollections throws Exception

### ISSUE THREAD

**Issue by hetzner** [api-ecosystem-for-laravel/dingo-api#26]
| Q                 | A
| ----------------- | ---
| Bug?              | yes
| New Feature?      | no
| Framework         | Laravel
| Framework version | 9.27.0
| Package version   | 4.0.1
| PHP version       | 8.1.2

TypeError: Dingo\Api\Routing\UrlGenerator::setRouteCollections(): Argument #1 ($collections) must be of type array, Illuminate\Support\Collection given, called in .../vendor/api-ecosystem-for-laravel/dingo-api/src/Provider/RoutingServiceProvider.php on line 55 and defined in .../vendor/api-ecosystem-for-laravel/dingo-api/src/Routing/UrlGenerator.php:46

**Comment by christoph-kluge** [comment]
Hello @hetzner thank you for reporting this. 

Could you add the whole stack trace so I could reproduce and test it?

Relates to #24.

**Comment by hetzner** [comment]
Hi @christoph-kluge 

here we go:

```
Dingo\Api\Routing\UrlGenerator::setRouteCollections(): Argument #1 ($collections) must be of type array, Illuminate\Support\Collection given, called in [...]/vendor/api-ecosystem-for-laravel/dingo-api/src/Provider/RoutingServiceProvider.php on line 55 {"userId":1,"exception":"[object] (TypeError(code: 0): Dingo\\Api\\Routing\\UrlGenerator::setRouteCollections(): Argument #1 ($collections) must be of type array, Illuminate\\Support\\Collection given, called in [...]/vendor/api-ecosystem-for-laravel/dingo-api/src/Provider/RoutingServiceProvider.php on line 55 at [...]/vendor/api-ecosystem-for-laravel/dingo-api/src/Routing/UrlGenerator.php:46)
[stacktrace]
#0 [...]/vendor/api-ecosystem-for-laravel/dingo-api/src/Provider/RoutingServiceProvider.php(55): Dingo\\Api\\Routing\\UrlGenerator->setRouteCollections()
#1 [...]/vendor/laravel/framework/src/Illuminate/Container/Container.php(871): Dingo\\Api\\Provider\\RoutingServiceProvider->Dingo\\Api\\Provider\\{closure}()
#2 [...]/vendor/laravel/framework/src/Illuminate/Container/Container.php(756): Illuminate\\Container\\Container->build()
#3 [...]/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(857): Illuminate\\Container\\Container->resolve()
#4 [...]/vendor/laravel/framework/src/Illuminate/Container/Container.php(692): Illuminate\\Foundation\\Application->resolve()
#5 [...]/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(842): Illuminate\\Container\\Container->make()
#6 [...]/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php(119): Illuminate\\Foundation\\Application->make()
#7 [...]/app/Http/Controllers/Api/v1/IdenfyServiceCallController.php(199): app()
#8 [...]/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): App\\Http\\Controllers\\Api\\v1\\IdenfyServiceCallController->create()
#9 [...]/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(45): Illuminate\\Routing\\Controller->callAction()
#10 [...]/vendor/laravel/framework/src/Illuminate/Routing/Route.php(261): Illuminate\\Routing\\ControllerDispatcher->dispatch()
#11 [...]/vendor/laravel/framework/src/Illuminate/Routing/Route.php(204): Illuminate\\Routing\\Route->runController()
#12 [...]/vendor/laravel/framework/src/Illuminate/Routing/Router.php(725): Illuminate\\Routing\\Route->run()
#13 [...]/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}()
#14 [...]/vendor/laravel/framework/src/Illuminate/Auth/Middleware/Authenticate.php(44): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#15 [...]/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Auth\\Middleware\\Authenticate->handle()
#16 [...]/vendor/api-ecosystem-for-laravel/dingo-api/src/Http/Middleware/PrepareController.php(43): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#17 [...]/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Dingo\\Api\\Http\\Middleware\\PrepareController->handle()
#18 [...]/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#19 [...]/vendor/laravel/framework/src/Illuminate/Routing/Router.php(726): Illuminate\\Pipeline\\Pipeline->then()
#20 [...]/vendor/laravel/framework/src/Illuminate/Routing/Router.php(703): Illuminate\\Routing\\Router->runRouteWithinStack()
#21 [...]/vendor/laravel/framework/src/Illuminate/Routing/Router.php(667): Illuminate\\Routing\\Router->runRoute()
#22 [...]/vendor/laravel/framework/src/Illuminate/Routing/Router.php(656): Illuminate\\Routing\\Router->dispatchToRoute()
#23 [...]/vendor/api-ecosystem-for-laravel/dingo-api/src/Routing/Adapter/Laravel.php(86): Illuminate\\Routing\\Router->dispatch()
#24 [...]/vendor/api-ecosystem-for-laravel/dingo-api/src/Routing/Router.php(496): Dingo\\Api\\Routing\\Adapter\\Laravel->dispatch()
#25 [...]/vendor/api-ecosystem-for-laravel/dingo-api/src/Http/Middleware/Request.php(123): Dingo\\Api\\Routing\\Router->dispatch()
#26 [...]/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Dingo\\Api\\Http\\Middleware\\Request->Dingo\\Api\\Http\\Middleware\\{closure}()
#27 [...]/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#28 [...]/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#29 [...]/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle()
#30 [...]/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#31 [...]/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#32 [...]/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle()
#33 [...]/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#34 [...]/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle()
#35 [...]/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#36 [...]/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle()
#37 [...]/vendor/fruitcake/laravel-cors/src/HandleCors.php(52): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#38 [...]/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Fruitcake\\Cors\\HandleCors->handle()
#39 [...]/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#40 [...]/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Fideloper\\Proxy\\TrustProxies->handle()
#41 [...]/app/Http/Middleware/ApiLog.php(23): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#42 [...]/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): App\\Http\\Middleware\\ApiLog->handle()
#43 [...]/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#44 [...]/vendor/api-ecosystem-for-laravel/dingo-api/src/Http/Middleware/Request.php(124): Illuminate\\Pipeline\\Pipeline->then()
#45 [...]/vendor/api-ecosystem-for-laravel/dingo-api/src/Http/Middleware/Request.php(101): Dingo\\Api\\Http\\Middleware\\Request->sendRequestThroughRouter()
#46 [...]/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Dingo\\Api\\Http\\Middleware\\Request->handle()
#47 [...]/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#48 [...]/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(142): Illuminate\\Pipeline\\Pipeline->then()
#49 [...]/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(111): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
#50 [...]/public/index.php(52): Illuminate\\Foundation\\Http\\Kernel->handle()
#51 {main}
```

The following code triggers the error:
```
app('Dingo\Api\Routing\UrlGenerator')->version('v1')->route('my.route')
```

And this is the route definition:
```
$api->post('my/path', ['as' => 'my.route',
                       'uses' => 'App\Http\Controllers\Api\v1\MyController@update']);
```

**Comment by christoph-kluge** [comment]
@hetzner thanks for providing details to reproduce it. 

The fix was merged with #24 and released via [v4.0.2](https://github.com/api-ecosystem-for-laravel/dingo-api/releases/tag/v4.0.2).
